Automattic/woocommerce-payments
# Fix Bug: Prevent Multi-Currency Conversion During REST API Request

Fixes #9181 

#### Changes proposed in this Pull Request









The changes in this PR ensure that the REST API requests are idempotent. This is accomplished by preventing multi-currency from converting the currency and price based on the user's session. Previously, if a shopper updated their frontend currency with a currency-switcher block, the REST API requests from the frontend would also be unintentionally impacted.

#### Testing instructions





**Replicate bug: Updating the frontend currency impacts the REST API requests**
* Switch to the `develop` branch.
* As a merchant, enable multi-currency in WooPayments and add at least one extra currency (e.g. `GBP` (default) + `EUR`).
* Create a new product and note the price in the default currency (e.g. `GBP`) and the ID.
* Add a "Currency Switcher" block widget ( **Appearance > Widgets**).
* As a shopper, navigate to the merchant shop and use the "Currency Switcher" to switch to the non-default currency (`EUR`).
* Make a REST API request to `https://merchant.com/wp-json/wc/v2/products/<product_id>?consumer_key=<key>&consumer_secret=<secret>` in the browser's search bar.
* Confirm that the price has changed and that the currency code reads `&euro;`.

**Confirm fix: Updating the frontend currency does not impact the REST API requests**
* Switch to the `fix/9181-unintended-mccy-conversion-during-rest-api-request` branch.
* As a merchant, enable multi-currency in WooPayments and add at least one extra currency (e.g. `GBP` (default) + `EUR`).
* Create a new product and note the price in the default currency (e.g. `GBP`) and the ID.
* Add a "Currency Switcher" block widget ( **Appearance > Widgets**).
* As a shopper, navigate to the merchant shop and use the "Currency Switcher" to switch to the non-default currency (`EUR`).
* Make a REST API request to `https://merchant.com/wp-json/wc/v2/products/<product_id>?consumer_key=<key>&consumer_secret=<secret>` in the browser's search bar.
* Confirm that the price has _not_ changed and that the currency code reads `&pound;` (default).
* Confirm that the correct price and currency appear in the `price_html` field.

**Ensure compatibility: Updating the frontend currency does not impact the REST API requests for compatible plugins**
* Switch to the `fix/9181-unintended-mccy-conversion-during-rest-api-request` branch.
* As a merchant, enable multi-currency in WooPayments and add at least one extra currency (e.g. `GBP` (default) + `EUR`).
* Using a [compatible plugin from this list](https://github.com/Automattic/woocommerce-payments/blob/4c417bca2ee072ddb342bcd9e0d85c0055ed3dd0/includes/multi-currency/Compatibility.php#L59-L67):
  * Create a new product (booking, subscription, etc.) and note the price in the default currency (e.g. `GBP`) and the ID.
* Add a "Currency Switcher" block widget ( **Appearance > Widgets**).
* As a shopper, navigate to the merchant shop and use the "Currency Switcher" to switch to the non-default currency (`EUR`).
* Make a REST API request to `https://merchant.com/wp-json/wc/v2/products/<product_id>?consumer_key=<key>&consumer_secret=<secret>` in the browser's search bar.
* Confirm that the price has _not_ changed and that the currency code reads `&pound;` (default).
* Confirm that the correct price and currency appear in the `price_html` field.

**Ensure unit tests all pass**
* Run the following command `npm run test:php`.
* Ensure all tests pass.

-------------------


**Post merge**
